### PR TITLE
First task done

### PR DESCRIPTION
**Pull request by artemohanjanyan**

### CODE REVIEW

File: math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            }
+        })
+
+        redraw()
+    }
+
+    fun redraw() {
+        data.forEach(DataTable::clear)
+
+        val height = if (plot.bounds.height == 0.0) bounds.height.toDouble() else plot.bounds.height
```
Comment (qradimir) [inline math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt:96]: Looks like a dirty hack. To review later

File: math/src/main/kotlin/ru/ifmo/ctddev/compmethods/ComputationMethods.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    : Iterator<Complex> {
+
+    private var cur = start
+
+    val current: Complex
+        get() = cur
+
+    private var next = computeNext()
+
+    override fun next(): Complex {
```
Comment (qradimir) [inline math/src/main/kotlin/ru/ifmo/ctddev/compmethods/ComputationMethods.kt:16]: First next() call should return start value, am I right?

File: math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                val x = (e?.x ?: return)
+                val y = (e?.y ?: return)
+
+                userData.clear()
+
+                val plotX = xAxisRenderer.viewToWorld(xAxis, x.toDouble(), true).toDouble()
+                val plotY = yAxisRenderer.viewToWorld(yAxis, plot.bounds.height - y.toDouble(), true).toDouble()
+
+                val seq = IterativeComputationSequence(Complex(plotX, plotY), MY_FUNC, MY_FUNC_LAMBDA, EPS)
+                userData.add(plotX, plotY)
```
Comment (qradimir) [inline math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt:82]: This line will be useless if IterativeComputationSequence returns start value at first next() call.

File: math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+val EPS = 1E-06
+
+fun createDataTable() = DataTable(Double::class.javaObjectType, Double::class.javaObjectType)
+
+class SinPlot() : JFrame() {
+
+    private val defaultWidth = 6.0
+
+    private var plot: XYPlot
```
Comment (qradimir) [inline math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt:31]: plot should be a value, not variable.

**Review by qradimir — COMMENTED** [review]
OK in general

File: math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt (hunk 3 above)
Comment (artemohanjanyan) [reply math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt:82]: Indeed

File: math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt (hunk 1 above)
Comment (artemohanjanyan) [reply math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt:96]: It works

Comment (qradimir) [reply math/src/main/kotlin/ru/ifmo/ctddev/compmethods/Newton.kt:96]: but how?

**Pull request merged**